src/uint.rs

```diff
@@ -1070,28 +1212,46 @@ where
 pub trait GetBit<I> {
     #[allow(missing_docs)]
     type Output;
+
+    #[doc(hidden)]
+    fn get_bit<IM: InternalMarker>(&self, &I) -> Self::Output;
 }
 
 #[allow(missing_docs)]
 pub type GetBitOut<N, I> = <N as GetBit<I>>::Output;
 
 // Base case
-impl<Un, Bn> GetBit<U0> for UInt<Un, Bn> {
+impl<Un, Bn> GetBit<U0> for UInt<Un, Bn>
+where
+    Bn: Bit
+{
     type Output = Bn;
+
+    fn get_bit<IM: InternalMarker>(&self, _: &U0) -> Self::Output {
+        self.lsb
+    }
```

Comment:
The `Bn: Bit` bound lets us copy `lsb` out of `&self`. It's possible that this bound could impact type checking performance. My gut says its unlikely, since determining whether a `T` is a `Bit` is just a matter of `T=B0` or `T=B`.